renderdoc/serialise/codecs/vk_cpp_codec/ext_object.h

```diff
@@ -0,0 +1,190 @@
+/******************************************************************************
+* The MIT License (MIT)
+*
+* Copyright (c) 2018 Baldur Karlsson
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in
+* all copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+* THE SOFTWARE.
+******************************************************************************/
+#pragma once
+
+#include <assert.h>
+#include <math.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <array>
+#include <list>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include "common/common.h"
+#include "core/core.h"
+#include "core/intervals.h"
+#include "serialise/rdcfile.h"
```

Comment:
The filename `ext_object.h` no longer makes sense, this would be better as `vk_cpp_helpers.h` or `vk_cpp_common.h` or something like that.

These includes can also be tidied up - remove any that aren't used anymore.

Reply:
Done

Reply:
I am not sure if the comments in github will survive a file rename, so I'll rename the file when all other comments are addressed, if there are no objections?

Reply:
Comments _should_ persist in the 'conversation' tab though they won't be visible in-line on the 'files changed' tab. If you want to put up a draft of with everything else before renaming though that's fine with me.

Reply:
I think that would be best.
Also, do you still want it inside codecs\vk_cpp_codec folder? My original thought was to place all vk cpp codec code there, but since most of it will go into the driver directory, and ext_object gets renamed to vk_cpp_codec_helper.h I might as well move it to the codecs directory. Or would you prefer codecs\vk_cpp_codec\vk_cpp_codec_helper.h?.. Please let me know.

Reply:
I'm not too fussed right now, I think this file stands to be refactored a bit at some point in the future anyway. I mentioned before that ultimately I'd want to separate out stuff that's not vulkan specific - which is actually most of what remains apart from `CanonicalUnionBranch`. I am already eyeing that one up to potentially fold into the structured data itself with an annotation at the point of serialise.

For now though it's fine to either leave it in the vk_cpp_codec folder or move it into vulkan driver. I can defer this kind of reorganisation stuff until there's actually another C++ codec being developed. The only thing I don't want to have is code from outside the drivers poking into driver internals, so anything that has a hard dependency on driver files must be within the driver folder.

Reply:
Cool, renamed it to vk_cpp_codec_common.h.